doc/02DMEThemeP.txt

```diff
@@ -0,0 +1,109 @@
+# DME Theme Protocol
+
+## Intro
+Each Theme must be created in the way described below to be correctly understood and applied by the DME-Core when the DME-Core calls the Themeing Task.
+
+Each Theme is identified by a directory called with the Theme name.
+The directory must presents the follwoing structure:
+```
+Theme Name
+├── Apps
+├── Backgrounds
+├── Fonts
+├── Palette
+├── Pointers
+├── Preview
+├── Shutdown
+├── Sounds
+└── Startup
+```
+
+## Protocol details
+Each theme must contain a cnfiguration file either called Choices or Config and the syntax is like the following:
+
+Protocol Example:
+```
+# Theme data header
+Name:The Theme
+Info:A theme
+Author:The one and only me
+License: Apache 2.0
+Release:1.0.0
+Notes: This theme requires at least 32K display or better
```

Comment:
Should there or should there not be a space after `:`? Does it matter?

Reply:
So the parser I am writing (and partially already wrote) doesn't care about the spaces, but you're right I should describe in more details the protocol form

Reply:
If both are allowed, I guess it is good that both variants are in the example. I suggest that the parser specification is hosted with the parser, and that this document merely refers to it, when it exists.

Reply:
Yup, so I still need to start documenting the parser, but it's pretty BASIC ;)

It parse the file linearly and (at the moment) has no support for option grouping (the "[ option block name ]").
It uses the ":" to split the option from its values and then trim both, I am also adding lower casing the token to make the theme file case insensitive at least for the tokens.

The reason why the parser is minimalistic is because I want to keep compatibility with The Real Small Software Company !Skinny app so people can create Themes using that app instead of having to do it manually. But the parser will support more options than what Skinny offers now. Maybe we can contact RSSC to improve skinny or something for the future.